File: services/api/app/schemas/kompass.py
"""Datenformen des Kompasses.

Eigene Schemas statt geliehener: Hier gibt es keine ``case_id`` als Pflichtfeld und keine
Fallbezüge im Rückgabewert. Wer später ein Feld aus dem Fallbereich hereinkopiert, soll das
merken, statt es zu erben.
"""
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field

from app.services.kompass_katalog import (
    SATZ_MAX_ZEICHEN,
    SCHRITT_MAX_ZEICHEN,
    VORHABEN_MAX_TITEL,
)
from app.services.kompass_portrait_service import MAX_ZEICHEN as PORTRAIT_MAX_ZEICHEN


class PulsCreate(BaseModel):
    """Ein Moment. Nur ``zustand`` ist Pflicht — das ist das Versprechen des Raums."""
    zustand: int = Field(ge=1, le=5)
    anspannung: int | None = Field(default=None, ge=0, le=10)
    worte: list[str] = Field(default_factory=list, max_length=10)
    notiz: str | None = Field(default=None, max_length=2000)
    #: „Was hat heute geholfen?" — wird nur bei guten Zuständen gefragt.
    geholfen: str | None = Field(default=None, max_length=500)
    #: „Das war mit …" — freiwillig, und nur ein eigener Fall.
    case_id: UUID | None = None


class Puls(BaseModel):
    id: UUID
    zustand: int
    zustand_label: str | None = None
    anspannung: int | None = None
    worte: list[str] = []
    notiz: str | None = None
    geholfen: str | None = None
    case_id: UUID | None = None
    created_at: datetime


class Krisenplan(BaseModel):
    """Der Plan für den Ernstfall.

    ``inhalt`` ist bewusst frei geformt: Die Abschnitte stehen im Katalog, nicht im
    Schema. Ein neuer Abschnitt ist damit ein Eintrag im Katalog und keine Migration.
    """
    inhalt: dict[str, Any] = Field(default_factory=dict)
    updated_at: datetime | None = None


class KrisenplanUpdate(BaseModel):
    inhalt: dict[str, Any] = Field(default_factory=dict)


# ── Die Sätze über mich ─────────────────────────────────────────────────────
#
# **Warum ``art`` und ``stand`` hier schlichte Zeichenketten sind und keine Literale.**
# Die erlaubten Wörter stehen im Katalog und in der Bedingung der Tabelle. Ein Literal
# hier wäre eine DRITTE Stelle, die mitwandern muss — und genau diese Sorte Fehler hat in
# dieser Codebasis schon dreimal zugeschlagen. Geprüft wird im Dienst gegen den Katalog,
# abgeglichen wird vom Wächter gegen die Datenbank.


class SatzCreate(BaseModel):
    art: str
    text: str = Field(min_length=1, max_length=SATZ_MAX_ZEICHEN)
    #: Woher der Satz kommt. Die Herkunft wird daraus abgeleitet, nicht mitgeschickt —
    #: sonst könnte jemand „von Echo vorgeschlagen" behaupten, was er selbst getippt hat.
    szene_id: UUID | None = None
    puls_id: UUID | None = None


class SatzUpdate(BaseModel):
    """Alles freiwillig — was fehlt, bleibt, wie es war."""
    text: str | None = Field(default=None, max_length=SATZ_MAX_ZEICHEN)
    art: str | None = None
    stand: str | None = None
    angeheftet: bool | None = None


class Satz(BaseModel):
    id: UUID
    art: str
    art_label: str | None = None
    text: str
    stand: str
    herkunft: str
    szene_id: UUID | None = None
    puls_id: UUID | None = None
    angeheftet: bool = False
    #: Woran Echo den Vorschlag festmacht. Nur bei Vorschlägen gesetzt — wer einen Satz
    #: selbst schreibt, muss sich nicht belegen.
    grund: str | None = None
    created_at: datetime
    #: Wann zugestimmt wurde. Gehört sichtbar in die Oberfläche: Ein Satz von vor zwei
    #: Jahren ist etwas anderes als einer von gestern.
    bestaetigt_at: datetime | None = None
    updated_at: datetime


class VorschlagsEntscheidung(BaseModel):
    """Ja oder nein zu einem Vorschlag — mehr gibt es hier nicht zu sagen."""
    annehmen: bool


class VorschlagsLauf(BaseModel):
    """Was ein Lauf ergeben hat.

    ``hinweis`` ist der Satz an die Person, wenn nichts dabei war — „zu wenig Material",
    „liegt noch etwas offen". Eine leere Liste ohne Erklärung sähe aus wie ein Fehler.
    """
    vorschlaege: list[Satz] = []
    hinweis: str | None = None


# ── Die Vorhaben ────────────────────────────────────────────────────────────


class Schritt(BaseModel):
    """Ein Schritt eines Vorhabens.

    ``id`` darf beim Anlegen fehlen — der Dienst vergibt dann eine. Ohne Kennung ließe
    sich ein Schritt nur über seine Position ansprechen, und beim Umsortieren hakte man
    den falschen ab.
    """
    id: str | None = None
    text: str = Field(min_length=1, max_length=SCHRITT_MAX_ZEICHEN)
    #: ISO-Zeitstempel oder None. Kommt vom Client zurück, wie er ihn bekommen hat.
    erledigt_at: str | None = None


class VorhabenCreate(BaseModel):
    """Nur der Titel ist Pflicht. Wer sich etwas vornimmt, weiß oft noch nicht, wie."""
    titel: str = Field(min_length=1, max_length=VORHABEN_MAX_TITEL)
    warum: str | None = Field(default=None, max_length=1000)
    schritte: list[Schritt] = Field(default_factory=list)
    rhythmus_tage: int = 0


class VorhabenUpdate(BaseModel):
    """Alles freiwillig — was fehlt, bleibt, wie es war."""
    titel: str | None = Field(default=None, max_length=VORHABEN_MAX_TITEL)
    warum: str | None = Field(default=None, max_length=1000)
    schritte: list[Schritt] | None = None
    rhythmus_tage: int | None = None
    stand: str | None = None
    #: Nur wenn jemand wirklich zurückgeschaut hat. Bei jeder Änderung mitzuschreiben
    #: hieße: Wer einen Tippfehler korrigiert, hat Rückschau gehalten.
    zurueckgeschaut: bool = False


class Vorhaben(BaseModel):
    id: UUID
    titel: str
    warum: str | None = None
    schritte: list[Schritt] = []
    #: Wie viele davon erledigt sind — gerechnet im Dienst, damit „2 von 5" überall
    #: dieselbe Auskunft ist.
    schritte_erledigt: int = 0
    rhythmus_tage: int = 0
    #: Wann zuletzt zurückgeschaut wurde. Ob daraus „fällig" folgt, entscheidet die
    #: Oberfläche — dieser Dienst hat bewusst keine Uhr.
    rueckschau_am: str | None = None
    stand: str
    stand_label: str | None = None
    created_at: datetime
    updated_at: datetime


# ── Die geführten Übungen ───────────────────────────────────────────────────


class UebungsSchritt(BaseModel):
    frage: str
    #: Für den Menschen. Geht NICHT in den Prompt — er enthält Beispiele.
    hinweis: str
    platzhalter: str


class Uebung(BaseModel):
    key: str
    label: str
    hinweis: str
    dauer: str
    #: ``satz`` oder ``vorhaben`` — was am Ende herauskommt.
    ergibt: str
    schritte: list[UebungsSchritt]


class UebungAbschluss(BaseModel):
    """Die Antworten, in der Reihenfolge der Fragen.

    Die Position IST die Zuordnung. Eine Antwort darf leer sein; unter
    ``MINDEST_ANTWORTEN`` gibt es kein Ergebnis.
    """
    antworten: list[str] = Field(default_factory=list, max_length=12)


class UebungsErgebnis(BaseModel):
    """Genau eines von beiden — oder keins, dann sagt ``hinweis`` warum.

    Das Ergebnis ist ein ENTWURF. Es gilt erst, wenn die Person zustimmt.
    """
    satz: Satz | None = None
    vorhaben: Vorhaben | None = None
    hinweis: str | None = None


# ── Das Selbstporträt ───────────────────────────────────────────────────────


class Portrait(BaseModel):
    id: UUID
    status: str
    text: str = ""
    created_at: datetime
    updated_at: datetime
    #: Wann es bestätigt wurde. Sichtbar — zwei Porträts nebeneinander sind nur dann
    #: eine Entwicklungsanzeige, wenn man weiß, aus welchen Monaten sie stammen.
    bestaetigt_at: datetime | None = None


class PortraitStand(BaseModel):
    """Was die Seite braucht — samt der Frage, ob ein neues entstehen darf."""
    entwurf: Portrait | None = None
    verlauf: list[Portrait] = []
    bereit: bool = False
    #: Warum nicht. Steht nur da, wenn ``bereit`` falsch ist.
    grund: str | None = None


class PortraitSichern(BaseModel):
    text: str = Field(min_length=1, max_length=PORTRAIT_MAX_ZEICHEN)


class PortraitVorschlag(BaseModel):
    """Echos Fassung — gespeichert ist damit noch nichts."""
    text: str = ""
    hinweis: str | None = None


# ── Deine Spur ──────────────────────────────────────────────────────────────


class SpurEreignis(BaseModel):
    """Ein Punkt auf der Zeitachse.

    ``art`` ist eine schlichte Zeichenkette und kein Literal: Die Liste steht im Dienst,
    und eine zweite hier waere die naechste Stelle, die mitwandern muss.
    """
    art: str
    am: datetime
    titel: str
    #: Ein kurzer Ausschnitt, nie der ganze Text. Die Achse ist eine Uebersicht.
    detail: str | None = None
    #: Nur bei Pulsen — faerbt den Punkt.
    zustand: int | None = None
    #: Wohin ein Klick fuehrt.
    ziel: str | None = None


class Belege(BaseModel):
    """Was seit dem Anfang eines Vorhabens dazugekommen ist.

    Ausdruecklich KEIN Prozentwert: ``zaehlung`` sagt, wie viel seitdem da ist, nicht
    wie weit jemand ist. Die Zuordnung liest die Person selbst.
    """
    seit: datetime
    zaehlung: dict[str, int] = Field(default_factory=dict)
    ereignisse: list[SpurEreignis] = []


class KompassUebersicht(BaseModel):
    """Was die Startseite braucht, in einem Zug."""
    letzter_puls: Puls | None = None
    verlauf: list[Puls] = []
    #: Wie viele Pulse in den letzten Wochen — eine Zahl, keine Serie.
    rhythmus: int = 0
    verlauf_tage: int = 28
    krisenplan_vorhanden: bool = False
    #: Wie viele bestätigte Sätze — für die Karte „Sätze über mich".
    saetze_bestaetigt: int = 0
    #: Wie viele Vorhaben gerade laufen.
    vorhaben_laufend: int = 0
    #: Ob gerade ein Selbstporträt entstehen darf — der Knopf auf der Startseite.
    portrait_bereit: bool = False
    #: Wie viele es schon gibt. Ohne diese Zahl liesse sich „noch keines, und noch nicht
    #: soweit" nicht von „eines da, gerade nicht fällig" unterscheiden — und die
    #: Startseite lüde zu einer Seite ein, die nur „jetzt nicht" sagt.
    portraits_anzahl: int = 0

File: apps/web/src/api/kompass.ts
/**
 * Mein Kompass — Puls, Verlauf, Krisenplan.
 *
 * **Der einzige Teil der Nutzer-API ohne Fall.** Alle anderen Aufrufe hier hängen unter
 * `/cases/{id}/…`; dieser nicht. Das ist das Versprechen des Raums: den eigenen Zustand
 * festhalten, ohne vorher einen Fall anzulegen.
 *
 * **Das Vokabular kommt vom Server.** Zustände, Wortfamilien und die Abschnitte des
 * Krisenplans stehen im Katalog des Backends, nicht hier. Sonst führt die Oberfläche eine
 * zweite Liste, die irgendwann von der ersten abweicht — und niemand merkt es, weil beide
 * für sich stimmig aussehen.
 */
import { apiClient } from './client'
import type { GbWortFamilie } from './gefuehlsbild'

/** Eine der fünf Stufen. Gespeichert wird `wert`, angezeigt `label`. */
export interface KompassZustand {
  wert: number
  label: string
  hinweis: string
}

export interface KompassAnspannung {
  label: string
  links: string
  rechts: string
  min: number
  max: number
}

export interface KrisenplanTeil {
  key: string
  label: string
  hinweis: string
  beispiel: string
}

/** Eine der sechs Arten, mit Erklärung und Beispiel — beides kommt vom Server. */
export interface SatzArt {
  key: string
  label: string
  hinweis: string
  beispiel: string
}

export interface SatzStand {
  key: string
  label: string
  hinweis: string
}

export interface VorhabenStand {
  key: string
  label: string
  hinweis: string
}

export interface RueckschauRhythmus {
  tage: number
  label: string
}

export interface KompassKatalog {
  zustaende: KompassZustand[]
  /** Ab dieser Stufe fragt der Puls, was geholfen hat. */
  guter_zustand_ab: number
  anspannung: KompassAnspannung
  /** Dieselben Wortfamilien wie im Gefühlsbild — bewusst, nicht zufällig. */
  wortfamilien: GbWortFamilie[]
  krisenplan_teile: KrisenplanTeil[]
  satz_arten: SatzArt[]
  /** Nur die drei sichtbaren. „Verworfen" ist kein Stand, den jemand wählt. */
  satz_staende: SatzStand[]
  satz_max_zeichen: number
  vorhaben_staende: VorhabenStand[]
  rueckschau_rhythmen: RueckschauRhythmus[]
  vorhaben_max_titel: number
  schritt_max_zeichen: number
  max_schritte: number
}

export interface Puls {
  id: string
  zustand: number
  zustand_label: string | null
  anspannung: number | null
  worte: string[]
  notiz: string | null
  geholfen: string | null
  case_id: string | null
  created_at: string
}

/** Nur `zustand` ist Pflicht. Alles andere darf fehlen — das ist der ganze Punkt. */
export interface PulsNeu {
  zustand: number
  anspannung?: number | null
  worte?: string[]
  notiz?: string | null
  geholfen?: string | null
  case_id?: string | null
}

export interface Krisenplan {
  /** Abschnitt → Zeilen. Die Schlüssel stehen im Katalog, nicht in diesem Typ. */
  inhalt: Record<string, unknown>
  updated_at: string | null
}

/**
 * Ein Satz über die eigene Person.
 *
 * **`bestaetigt_at` gehört sichtbar in die Oberfläche.** Ein bestätigter Satz ist eine
 * Selbsteinschätzung von dem Tag, an dem jemand zugestimmt hat — kein Befund und keine
 * Eigenschaft. Ohne das Datum liest er sich wie eine Tatsache über einen Menschen.
 */
export interface Satz {
  id: string
  art: string
  art_label: string | null
  text: string
  stand: 'entwurf' | 'bestaetigt' | 'ueberholt'
  herkunft: 'selbst' | 'szene' | 'puls' | 'echo'
  szene_id: string | null
  puls_id: string | null
  angeheftet: boolean
  /**
   * Woran Echo den Vorschlag festmacht — „Das kam in drei Situationen vor".
   *
   * Nur bei Vorschlägen gesetzt. Er gehört sichtbar neben den Satz: Ohne Beleg ist
   * Zustimmen ein Raten, und ein Vorschlag über die eigene Person will überlegt werden.
   */
  grund: string | null
  created_at: string
  bestaetigt_at: string | null
  updated_at: string
}

/** Die Herkunft wird abgeleitet, nicht mitgeschickt — sie lässt sich nicht behaupten. */
export interface SatzNeu {
  art: string
  text: string
  szene_id?: string | null
  puls_id?: string | null
}

export interface SatzAenderung {
  text?: string
  art?: string
  stand?: 'entwurf' | 'bestaetigt' | 'ueberholt'
  angeheftet?: boolean
}

/** Was ein Lauf ergeben hat. `hinweis` erklärt eine leere Liste, statt sie zu lassen. */
export interface VorschlagsLauf {
  vorschlaege: Satz[]
  hinweis: string | null
}

/** Ein Schritt eines Vorhabens. `id` fehlt nur, solange er noch nicht gespeichert ist. */
export interface Schritt {
  id?: string | null
  text: string
  erledigt_at?: string | null
}

export interface Vorhaben {
  id: string
  titel: string
  warum: string | null
  schritte: Schritt[]
  schritte_erledigt: number
  rhythmus_tage: number
  /** Wann zuletzt zurückgeschaut wurde. Ob daraus „fällig" folgt, rechnet `lib/kompass`. */
  rueckschau_am: string | null
  stand: 'laufend' | 'erreicht' | 'ruht'
  stand_label: string | null
  created_at: string
  updated_at: string
}

export interface VorhabenNeu {
  titel: string
  warum?: string | null
  schritte?: Schritt[]
  rhythmus_tage?: number
}

export interface VorhabenAenderung {
  titel?: string
  warum?: string | null
  schritte?: Schritt[]
  rhythmus_tage?: number
  stand?: 'laufend' | 'erreicht' | 'ruht'
  /** Nur setzen, wenn wirklich zurückgeschaut wurde — sonst ist der Rhythmus wertlos. */
  zurueckgeschaut?: boolean
}

/** Eine Frage einer geführten Übung. Der Hinweis ist für dich, nicht für Echo. */
export interface UebungsSchritt {
  frage: string
  hinweis: string
  platzhalter: string
}

export interface Uebung {
  key: string
  label: string
  hinweis: string
  dauer: string
  /** Was am Ende herauskommt: `satz` oder `vorhaben`. */
  ergibt: 'satz' | 'vorhaben'
  schritte: UebungsSchritt[]
}

/**
 * Was eine Übung ergeben hat — genau eines von beiden, oder keins.
 *
 * Das Ergebnis ist ein ENTWURF: Es liegt schon im Kompass, gilt aber erst mit deiner
 * Zustimmung. Deshalb muss man es nicht sofort entscheiden.
 */
export interface UebungsErgebnis {
  satz: Satz | null
  vorhaben: Vorhaben | null
  hinweis: string | null
}

/**
 * Ein Selbstporträt — ein Entwurf oder eine datierte Momentaufnahme.
 *
 * Ein bestätigtes Porträt ist unveränderlich. Bearbeitet wird der Entwurf; das vom März
 * neben dem vom September zu lesen, ist die Entwicklungsanzeige.
 */
export interface Portrait {
  id: string
  status: 'entwurf' | 'bestaetigt'
  text: string
  created_at: string
  updated_at: string
  bestaetigt_at: string | null
}

export interface PortraitStand {
  entwurf: Portrait | null
  verlauf: Portrait[]
  /** Ob gerade ein neues entstehen darf. */
  bereit: boolean
  /** Warum nicht — steht nur da, wenn `bereit` falsch ist. */
  grund: string | null
}

/** Echos Fassung. Gespeichert ist damit noch nichts. */
export interface PortraitVorschlag {
  text: string
  hinweis: string | null
}

/** Ein Punkt auf der Zeitachse. */
export interface SpurEreignis {
  art: 'puls' | 'satz' | 'vorhaben' | 'schritt' | 'portrait' | 'szene'
  am: string
  titel: string
  /** Ein kurzer Ausschnitt, nie der ganze Text. */
  detail: string | null
  /** Nur bei Pulsen — färbt den Punkt. */
  zustand: number | null
  ziel: string | null
}

/**
 * Was seit dem Anfang eines Vorhabens dazugekommen ist.
 *
 * Ausdrücklich KEIN Prozentwert: `zaehlung` sagt, wie viel seitdem da ist, nicht wie
 * weit jemand ist. Ob es zusammengehört, liest die Person selbst.
 */
export interface Belege {
  seit: string
  zaehlung: Partial<Record<SpurEreignis['art'], number>>
  ereignisse: SpurEreignis[]
}

export interface KompassUebersicht {
  letzter_puls: Puls | null
  verlauf: Puls[]
  /** Wie viele Momente im Zeitraum — eine Zahl, keine Serie. */
  rhythmus: number
  verlauf_tage: number
  krisenplan_vorhanden: boolean
  saetze_bestaetigt: number
  vorhaben_laufend: number
  portrait_bereit: boolean
  /**
   * Wie viele es schon gibt.
   *
   * Ohne diese Zahl liesse sich „noch keines, und noch nicht soweit" nicht von „eines da,
   * gerade nicht fällig" unterscheiden — und die Startseite lüde zu einer Seite ein, die
   * nur „jetzt nicht" sagt.
   */
  portraits_anzahl: number
}

const basis = '/me/kompass'

export const kompassApi = {
  katalog: () =>
    apiClient.get<KompassKatalog>(`${basis}/katalog`).then(r => r.data),

  uebersicht: () =>
    apiClient.get<KompassUebersicht>(basis).then(r => r.data),

  pulsAnlegen: (puls: PulsNeu) =>
    apiClient.post<Puls>(`${basis}/puls`, puls).then(r => r.data),

  pulsLoeschen: (pulsId: string) =>
    apiClient.delete(`${basis}/puls/${pulsId}`).then(() => undefined),

  verlauf: (tage?: number) =>
    apiClient
      .get<Puls[]>(`${basis}/verlauf`, tage ? { params: { tage } } : undefined)
      .then(r => r.data),

  krisenplan: () =>
    apiClient.get<Krisenplan>(`${basis}/krisenplan`).then(r => r.data),

  krisenplanSpeichern: (inhalt: Record<string, string[]>) =>
    apiClient.put<Krisenplan>(`${basis}/krisenplan`, { inhalt }).then(r => r.data),

  saetze: () =>
    apiClient.get<Satz[]>(`${basis}/saetze`).then(r => r.data),

  satzAnlegen: (satz: SatzNeu) =>
    apiClient.post<Satz>(`${basis}/saetze`, satz).then(r => r.data),

  satzAendern: (satzId: string, aenderung: SatzAenderung) =>
    apiClient.patch<Satz>(`${basis}/saetze/${satzId}`, aenderung).then(r => r.data),

  satzLoeschen: (satzId: string) =>
    apiClient.delete(`${basis}/saetze/${satzId}`).then(() => undefined),

  /**
   * Echo liest die letzten Szenen und Momente und schlägt Sätze vor.
   *
   * Dauert länger als die übrigen Aufrufe — hier spricht ein Modell. Die Vorschläge
   * werden serverseitig als Entwürfe abgelegt; die Antwort ist nur die Abkürzung.
   */
  vorschlaegeHolen: () =>
    apiClient
      .post<VorschlagsLauf>(`${basis}/saetze/vorschlaege`, undefined, { timeout: 60_000 })
      .then(r => r.data),

  vorhaben: () =>
    apiClient.get<Vorhaben[]>(`${basis}/vorhaben`).then(r => r.data),

  vorhabenAnlegen: (v: VorhabenNeu) =>
    apiClient.post<Vorhaben>(`${basis}/vorhaben`, v).then(r => r.data),

  vorhabenAendern: (id: string, aenderung: VorhabenAenderung) =>
    apiClient.patch<Vorhaben>(`${basis}/vorhaben/${id}`, aenderung).then(r => r.data),

  vorhabenLoeschen: (id: string) =>
    apiClient.delete(`${basis}/vorhaben/${id}`).then(() => undefined),

  uebungen: () =>
    apiClient.get<Uebung[]>(`${basis}/uebungen`).then(r => r.data),

  /** Dauert länger als die übrigen Aufrufe — hier spricht ein Modell. */
  uebungAbschliessen: (schluessel: string, antworten: string[]) =>
    apiClient
      .post<UebungsErgebnis>(
        `${basis}/uebungen/${schluessel}/abschliessen`,
        { antworten },
        { timeout: 60_000 },
      )
      .then(r => r.data),

  vorschlagEntscheiden: (satzId: string, annehmen: boolean) =>
    apiClient
      .post<Satz>(`${basis}/saetze/${satzId}/entscheidung`, { annehmen })
      .then(r => r.data),

  /**
   * Alles Festgehaltene auf einer Achse, neueste zuerst.
   *
   * `szenen` ist aus, bis jemand es einschaltet: Szenen gehören zu Fällen, und der
   * Kompass ist der Raum ohne Fall.
   */
  spur: (tage: number, szenen: boolean) =>
    apiClient
      .get<SpurEreignis[]>(`${basis}/spur`, { params: { tage, szenen } })
      .then(r => r.data),

  /**
   * Die bestätigten Sätze, die aus dieser Szene gewachsen sind — der Rückverweis.
   *
   * Unter `/me/kompass` und nicht unter dem Fall: Was herauskommt, sind Sätze über die
   * Person, und die gehören in den Raum, der ihr gehört.
   */
  saetzeZuSzene: (szeneId: string) =>
    apiClient.get<Satz[]>(`${basis}/szenen/${szeneId}/saetze`).then(r => r.data),

  vorhabenBelege: (id: string) =>
    apiClient.get<Belege>(`${basis}/vorhaben/${id}/belege`).then(r => r.data),

  portrait: () =>
    apiClient.get<PortraitStand>(`${basis}/portrait`).then(r => r.data),

  /**
   * Echo schreibt einen Vorschlag — **speichert nichts**.
   *
   * Der längste Aufruf im Kompass: Es geht mehr Material hin und mehrere Absätze zurück.
   * Ist gerade kein neues Porträt fällig, kommt ein leerer Text mit `hinweis` zurück —
   * kein Fehler, sondern die Antwort auf die Frage.
   */
  portraitSchreiben: () =>
    apiClient
      .post<PortraitVorschlag>(`${basis}/portrait/schreiben`, undefined,
        { timeout: 120_000 })
      .then(r => r.data),

  portraitSichern: (text: string) =>
    apiClient.put<Portrait>(`${basis}/portrait`, { text }).then(r => r.data),

  portraitBestaetigen: () =>
    apiClient.post<Portrait>(`${basis}/portrait/bestaetigen`).then(r => r.data),

  portraitEntwurfVerwerfen: () =>
    apiClient.delete(`${basis}/portrait/entwurf`).then(() => undefined),
}
